explorations/exploring_recipes.Rmd

```diff
@@ -0,0 +1,265 @@
+# Train recipe
+trained_rec <- prep(standardized, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+Add more steps with by just adding to the trained recipe. Get a list of steps using `apropos("^step_")`
+```{r}
+apropos("^step_")
+```
+
+Updating a recipe
+```{r}
+# Update recipe
+trained_rec <- trained_rec %>%
+  step_nzv(all_predictors())
+
+# Train updated recipe (recall that you'll have to start with a fresh copy of
+# data_train to apply the recipe from the beginning).
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+trained_rec <- prep(trained_rec, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+Dummification
+```{r}
+# Update recipe
+trained_rec <- trained_rec %>%
+  step_dummy(starts_with("gender"))
+
+# Train updated recipe
+data_train <- data[train_index$Resample1, ]
+data_test <- data[-train_index$Resample1, ]
+trained_rec <- prep(trained_rec, training = data_train)
+
+# Apply recipe
+data_train <- bake(trained_rec, newdata = data_train)
+data_test  <- bake(trained_rec, newdata = data_test)
+```
+
+## Adding a custom step
+Factors should get a level called, `hcai_missing`. Then, NA values should be set to this value. If there are no missing values, this column will still generate a dummy column. To do this, we need a custom recipes function.
+
+```{r}
+# This is the user-facing function
+step_hcai_missing <- function(recipe, 
+                              ..., 
+                              role = NA, 
+                              trained = FALSE,
+                              na_percentage = NULL) {
+  terms <- rlang::quos(...) 
+  if (length(terms) == 0)
+    stop("Please supply at least one variable specification. See ?selections.")
+  add_step(
+    recipe, 
+    step_hcai_missing_new(
+      terms = terms,
+      role = role,
+      trained = trained,
+      na_percentage = na_percentage
+    )
+  )
+}
+
+# Initialze a new object
+step_hcai_missing_new <- function(terms = NULL, 
+                                  role = NA, 
+                                  trained = FALSE,
+                                  na_percentage = NULL) {
+  step(
+    subclass = "hcai_missing", 
+    terms = terms,
+    role = role,
+    trained = trained,
+    na_percentage = na_percentage
+  )
+}
+
+# prep method
+prep.step_hcai_missing <- function(x, training, info = NULL, ...) {
+  col_names <- terms_select(terms = x$terms, info = info)
+  na_percentage <- sapply(training[, col_names], function(x) {
+    100 * round(sum(is.na(x)) / length(x), 3)
+    }
+  )
+  
+  step_hcai_missing_new(
+    terms = x$terms,
+    role = x$role,
+    trained = TRUE,
+    na_percentage = na_percentage
+  )
+}
+
+# bake method
+bake.step_hcai_missing <- function(object, newdata, ...) {
+  require(tibble)
```

Comment:
This would add tibble functions to the users' search path, which we don't want to do. Instead `@import tibble` in the roxygen to have all its functions available to our functions.

Reply:
Fixed and documented how to do imports.